Zadania/zad5_false_sharing.cpp

```diff
@@ -0,0 +1,65 @@
+#include <array>
+#include <chrono>
+#include <cstddef>
+#include <functional>
+#include <iostream>
+#include <numeric>
+#include <thread>
+
+struct falseSharingData
+{
+    int firstValue, secondValue;
+
+    void operator++ ()
+    {
+        ++firstValue;
+        ++secondValue;
+    }
+
+    falseSharingData& operator= (int newValue)
+    {
+        firstValue += newValue;
+        secondValue += newValue;
+
+        return *this;
+    }
+};
+
+class falseSharingContainer
+{
+public:
+
+    falseSharingContainer()
+    {
+        std::iota(falseArray.begin(), falseArray.end(), 0);
+    }
+
+    void alternatingAddition(int i)
+    {
+        for(i; i < falseArray.size(); i += 4)
```

Comment:
Jest możliwe, że tutaj będzie false sharing, ale nie zawsze. Jeśli jeden wątek pójdzie bardziej do przodu niż drugi to nie będzie. Gdybyś nie inkrementował i tylko w pętli np milion razy inkrementował falseArray[0] na jednym wątku, a falseArray[1] na drugim to na pewno byłby false sharing, bo te dane będą leżeć obok siebie, a więc będą współdzielić linię cache'u.